index.js

```diff
@@ -982,6 +1036,39 @@ class Batch {
     }
   }
 
+  async links (key, seq) {
+    let node = await this.getRoot(false)
+    if (!node) return false
+
+    if (node.block.seq === seq) return true
+
+    while (true) {
+      if (node.block.isTarget(key)) return false
+
+      let s = 0
+      let e = node.keys.length
+      let c
+
+      while (s < e) {
+        const mid = (s + e) >> 1
+
+        if (node.keys[mid].seq === seq) return true
+        c = b4a.compare(key, await node.getKey(mid))
+
+        if (c === 0) return false
+
+        if (c < 0) e = mid
+        else s = mid + 1
+      }
+
+      if (!node.children.length) return false
+
+      const i = c < 0 ? e : s
+      node = await node.getChildNode(i)
+      if (node.block.seq === seq) return true
```

Comment:
Sanity check: no need for `if (!node) return false` before this? (if guaranteed by `node.children.length` I suppose it's ok, since this is a batch and so it's race-condition safe)

Reply:
It should always have the node here so better to crash